include/circt/Dialect/SystemC/SystemCStructure.td

```diff
@@ -11,8 +11,72 @@
 //
 //===----------------------------------------------------------------------===//
 
+def SCModuleOp : SystemCOp<"module", [
+    Symbol,
+    FunctionOpInterface,
+    IsolatedFromAbove,
+    SingleBlock,
+    NoTerminator,
+    RegionKindInterface,
+    DeclareOpInterfaceMethods<CallableOpInterface>,
+    DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmBlockArgumentNames"]>,
+    HasParent<"mlir::ModuleOp">
+  ]> {
+  let summary = "Define a SystemC SC_MODULE.";
+  let description = [{
+    Represents the SC_MODULE macro as described in IEEE 1666-2011 §5.2.5.
+    Models input, output and inout ports as module arguments (as opposed to
+    `sc_signal`s which are modeled by a separate `systemc.signal` operation),
+    but are nonetheless emitted as regular struct fields.
+  }];
+
+  let arguments = (ins PortDirectionsAttr:$portDirections, StrArrayAttr:$portNames);
+  let regions = (region SizedRegion<1>: $body);
+
+  let hasCustomAssemblyFormat = 1;
+  let hasVerifier = 1;
+
+  let extraClassDeclaration = [{
+    // Implement RegionKindInterface.
+    static RegionKind getRegionKind(unsigned index) { return RegionKind::Graph;}
```

Comment:
This lack of a space after`;` bothers me, but I also grok tha tthis is to keep it under 80 chars. Consider:
```suggestion
    static RegionKind getRegionKind(unsigned index) { 
      return RegionKind::Graph;
    }
```

Reply:
You might also want to update it in `HWStructure.td` then 😉